Svelvet (re-)compiling to src/ instead of dist/ on Windows, triggering babel parse errors

So, for some reason, running svelvet on windows causes the hot-reload triggered and re-compiled files to appear in src/[file].js rather than dist/[file.js]. This has the immediate effect of hot reload not working and the follow-up effect of creating extra (babelized/compiled) copies of the same file in .src that in turn get re-babelized again, it seems. 

This results in various errors, but the most common ones I've seen are:
`Cannot find module 'svelte/internal.js'` 
or
```
Failed to transform with babel: src\App.js
{ SyntaxError: unknown: Unterminated string constant (87:23)

  85 | }
  86 |
> 87 | export default App;nent", {
     |                        ^
  88 |       component: this,
  89 |       tagName: "App",
  90 |       options,
    at Parser.raise (...\node_modules\@babel\parser\lib\index.js:7017:17)
    at Parser.readString (...\node_modules\@babel\parser\lib\index.js:8010:20)
    at Parser.getTokenFromCode (...\node_modules\@babel\parser\lib\index.js:7664:14)
    at Parser.nextToken (...\node_modules\@babel\parser\lib\index.js:7241:12)
    at Parser.next (...\node_modules\@babel\parser\lib\index.js:7170:10)
    at Parser.parseIdentifierName (...\node_modules\@babel\parser\lib\index.js:10366:10)
    at Parser.parseIdentifier (...\node_modules\@babel\parser\lib\index.js:10335:23)
    at Parser.parseExprAtom (...\node_modules\@babel\parser\lib\index.js:9516:27)
    at Parser.parseExprSubscripts (...\node_modules\@babel\parser\lib\index.js:9259:23)
    at Parser.parseMaybeUnary (...\node_modules\@babel\parser\lib\index.js:9239:21)
  pos: 1796,
  loc: Position { line: 87, column: 23 },
  code: 'BABEL_PARSE_ERROR' }
```

(note that it looks like the end of the file just repeats - I'm guessing this is due to the src/App.js being babelized over the babelized src/App.svelte in dist/App.js or the other way around? eg. we're seeing the most of one file and the end of another file there?)

Steps basically are;
1. Have no dist/ and just App.svelte and index.html in src/
2. run svelvet, everything compiles to dist properly and dev server is started up;
```
$ svelvet
(node:15176) ExperimentalWarning: The fs.promises API is experimental
Copied asset dist/index.html
Svelte compiled dist/components/Header.js
Svelte compiled dist/App.js

Building web_modules with snowpack...
- snowpack installing...
√ snowpack installed: svelte/internal. [0.31s]

Babel transformed dist/components/Header.js
Babel transformed dist/App.js
Watching for files...
Server running on http://localhost:8080
```
3. edit App.svelte, log shows a lot of compilation happening (note src instead of dist);
```
Svelte compiled src\App.js
Babel transformed src\App.js
Svelte compiled src\App.js
Svelte compiled src\App.js
Babel transformed src\App.js
Babel transformed src\App.js
Svelte compiled src\App.js
Svelte compiled src\App.js
Babel transformed src\App.js
Babel transformed src\App.js
Svelte compiled src\App.js
Babel transformed src\App.js
Svelte compiled src\App.js
Babel transformed src\App.js
Svelte compiled src\App.js
Babel transformed src\App.js
```
4. no hot reload is seen, nothing is reloaded. src/App.js is created, containing the *new* compiled version
5. edit App.svelte again
6. babel compilation errors appear, subsequent log is as follows:
```
Svelte compiled src\App.js
Svelte compiled src\App.js
Babel transformed src\App.js

Failed to transform with babel: src\App.js
{ SyntaxError: unknown: Unterminated string constant (87:23)

  85 | }
  86 |
> 87 | export default App;nent", {
     |                        ^
  88 |       component: this,
  89 |       tagName: "App",
  90 |       options,
    at Parser.raise (...\node_modules\@babel\parser\lib\index.js:7017:17)
    at Parser.readString (...\node_modules\@babel\parser\lib\index.js:8010:20)
    at Parser.getTokenFromCode (...\node_modules\@babel\parser\lib\index.js:7664:14)
    at Parser.nextToken (...\node_modules\@babel\parser\lib\index.js:7241:12)
    at Parser.next (...\node_modules\@babel\parser\lib\index.js:7170:10)
    at Parser.parseIdentifierName (...\node_modules\@babel\parser\lib\index.js:10366:10)
    at Parser.parseIdentifier (...\node_modules\@babel\parser\lib\index.js:10335:23)
    at Parser.parseExprAtom (...\node_modules\@babel\parser\lib\index.js:9516:27)
    at Parser.parseExprSubscripts (...\node_modules\@babel\parser\lib\index.js:9259:23)
    at Parser.parseMaybeUnary (...\node_modules\@babel\parser\lib\index.js:9239:21)
  pos: 1802,
  loc: Position { line: 87, column: 23 },
  code: 'BABEL_PARSE_ERROR' }

Svelte compiled src\App.js

Failed to transform with babel: src\App.js
{ SyntaxError: unknown: Unterminated string constant (87:23)

  85 | }
  86 |
> 87 | export default App;nent", {
     |                        ^
  88 |       component: this,
  89 |       tagName: "App",
  90 |       options,
    at Parser.raise (...\node_modules\@babel\parser\lib\index.js:7017:17)
    at Parser.readString (...\node_modules\@babel\parser\lib\index.js:8010:20)
    at Parser.getTokenFromCode (...\node_modules\@babel\parser\lib\index.js:7664:14)
    at Parser.nextToken (...\node_modules\@babel\parser\lib\index.js:7241:12)
    at Parser.next (...\node_modules\@babel\parser\lib\index.js:7170:10)
    at Parser.parseIdentifierName (...\node_modules\@babel\parser\lib\index.js:10366:10)
    at Parser.parseIdentifier (...\node_modules\@babel\parser\lib\index.js:10335:23)
    at Parser.parseExprAtom (...\node_modules\@babel\parser\lib\index.js:9516:27)
    at Parser.parseExprSubscripts (...\node_modules\@babel\parser\lib\index.js:9259:23)
    at Parser.parseMaybeUnary (...\node_modules\@babel\parser\lib\index.js:9239:21)
  pos: 1802,
  loc: Position { line: 87, column: 23 },
  code: 'BABEL_PARSE_ERROR' }

Svelte compiled src\App.js
Svelte compiled src\App.js

Failed to transform with babel: src\App.js
{ SyntaxError: unknown: Unterminated string constant (87:23)

  85 | }
  86 |
> 87 | export default App;nent", {
     |                        ^
  88 |       component: this,
  89 |       tagName: "App",
  90 |       options,
    at Parser.raise (...\node_modules\@babel\parser\lib\index.js:7017:17)
    at Parser.readString (...\node_modules\@babel\parser\lib\index.js:8010:20)
    at Parser.getTokenFromCode (...\node_modules\@babel\parser\lib\index.js:7664:14)
    at Parser.nextToken (...\node_modules\@babel\parser\lib\index.js:7241:12)
    at Parser.next (...\node_modules\@babel\parser\lib\index.js:7170:10)
    at Parser.parseIdentifierName (...\node_modules\@babel\parser\lib\index.js:10366:10)
    at Parser.parseIdentifier (...\node_modules\@babel\parser\lib\index.js:10335:23)
    at Parser.parseExprAtom (...\node_modules\@babel\parser\lib\index.js:9516:27)
    at Parser.parseExprSubscripts (...\node_modules\@babel\parser\lib\index.js:9259:23)
    at Parser.parseMaybeUnary (...\node_modules\@babel\parser\lib\index.js:9239:21)
  pos: 1802,
  loc: Position { line: 87, column: 23 },
  code: 'BABEL_PARSE_ERROR' }


Failed to transform with babel: src\App.js
{ SyntaxError: unknown: Unterminated string constant (87:23)

  85 | }
  86 |
> 87 | export default App;nent", {
     |                        ^
  88 |       component: this,
  89 |       tagName: "App",
  90 |       options,
    at Parser.raise (...\node_modules\@babel\parser\lib\index.js:7017:17)
    at Parser.readString (...\node_modules\@babel\parser\lib\index.js:8010:20)
    at Parser.getTokenFromCode (...\node_modules\@babel\parser\lib\index.js:7664:14)
    at Parser.nextToken (...\node_modules\@babel\parser\lib\index.js:7241:12)
    at Parser.next (...\node_modules\@babel\parser\lib\index.js:7170:10)
    at Parser.parseIdentifierName (...\node_modules\@babel\parser\lib\index.js:10366:10)
    at Parser.parseIdentifier (...\node_modules\@babel\parser\lib\index.js:10335:23)
    at Parser.parseExprAtom (...\node_modules\@babel\parser\lib\index.js:9516:27)
    at Parser.parseExprSubscripts (...\node_modules\@babel\parser\lib\index.js:9259:23)
    at Parser.parseMaybeUnary (...\node_modules\@babel\parser\lib\index.js:9239:21)
  pos: 1802,
  loc: Position { line: 87, column: 23 },
  code: 'BABEL_PARSE_ERROR' }
```

Will probably be investigating further and happy to help out in debugging - I'm rather excited about the project and idea!

--- a/src/index.ts
+++ b/src/index.ts
@@ -75,9 +75,7 @@ async function compile(
             newSource = result.js.code;
         }
 
-        const destPath = srcPath
-            .replace(/^src\//, 'dist/')
-            .replace(/.svelte$/, '.js');
+        const destPath = getDestPath(srcPath).replace(/.svelte$/, '.js');
         // Create all ancestor directories for this file
         await fs.mkdir(path.dirname(destPath), { recursive: true });
         await fs.writeFile(destPath, newSource);
@@ -101,11 +99,15 @@ async function compile(
 }
 
 async function copyFile(srcPath: string): Promise<void> {
-    const destPath = srcPath.replace(/^src\//, 'dist/');
+    const destPath = getDestPath(srcPath);
     // Create all ancestor directories for this file
     await fs.mkdir(path.dirname(destPath), { recursive: true });
     await fs.copyFile(srcPath, destPath);
     console.info(`Copied asset ${destPath}`);
+}
+
+function getDestPath(srcPath: string): string {
+    return srcPath.replace(new RegExp(`^src\\${path.sep}`), `dist${path.sep}`);
 }
 
 // Update the import paths to correctly point to web_modules.

